Date input and switch item refactor

mercedesb opened the pull request:
## JIRA

* N/A

<!--
## Technical Design Doc
- link to TDD if the feature had one (optional)
-->

## Description

* rename Table icon to TableIcon to prevent collisions with html table element
* add slot and srOnlyLabel to SwitchItem to support rendering icons instead of text in some cases
* add DateInput to ui-components so other teams can use without copypasta
* fix storybook docs and story rendering issues

<!--
## Screenshots
- before and after screenshots for features with visual changes
-->

<!--
## Tests
- info about what was used to validate the feature (more for larger, more complicated features)
-->

<!--
## Areas of Concern
- to call out what could be a problem (optional)
-->

<!--
## Questions
- to ask any questions for the reviewers (optional)
-->

<!--
## GIFs/Memes
- (optional)
-->

## Reviewer Checklist

This section is to be filled out by reviewers

### Testing

- [x] This code was tested by somebody other than the developer. Do not merge until this has been done.

bethqiang commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@lob/ui-components",
-  "version": "0.0.83",
+  "version": "0.0.84",
```
Nit: should be 86 now ☺️

bethqiang commented on src/components/DateInput/DateInput.mdx:
```diff
@@ -0,0 +1,54 @@
+
+## Props
+
+<ArgsTable story={PRIMARY_STORY} />
```
Great docs! Felt like this was very thorough.

bethqiang commented on src/components/Switch/SwitchGroup.stories.js:
```diff
@@ -33,3 +34,24 @@ Primary.args = {
+  setup: () => ({ args }),
+  template: `
+    <switch-group v-bind="args">
+      <switch-item name='chart' label="Map view" value='map' v-model="withIconsModel" sr-only-label><earth class="w-6 h-6" /></switch-item>
```
Nits: inconsistencies between single/double quotes, a little hard to read when it's all on one line

bethqiang reviewed (approved):
Looks great to me! The only general things I'd comment on is:
* I'm not seeing a way to open the datepicker with a keyboard only (I tried with both space and enter keys)
* not all of the knobs work completely (e.g. open doesn't do anything, trying to change min/max throws an error) but (1) I'm sure this is probably the case for a few of our components and (2) I'm really very ok with pushing that off until after the replatform and before we publicize our library

mercedesb commented:
@bethqiang We should probably remove controls for all of our `v-model` props since we can't mutate them from the controls (since they are defined in the js of the story itself). I went ahead and did that for this one. And I added a work around for `min` and `max`. Storybook handles dates super weird and [there's an open issue](https://github.com/storybookjs/storybook/issues/11822)

Good catch on keyboard navigability, I added a space keydown handler (seemed a better choice than enter in the UX b/c enter does so much stuff when the thing is open)
